### ISSUE TITLE
Flymake/Flycheck is problematic after toggle comment keyboard shortcut

### ISSUE THREAD

**Issue by cello-phane** [quick-lint/quick-lint-js#1014]
![emacs_tPrJL3rX9z](https://github.com/quick-lint/quick-lint-js/assets/87787413/80cb4e94-48a4-4c97-b942-1b740bc3b737)

In the gif above, I manually remove and insert a `//`, and everything works.
As soon as a comment is toggled with the emacs `Cx-C;` kbd shortcut, the line is not updated, and the error message persists. (Also the buffer doesn't list the error until I use the kb shortcut, which could be the root of the problem with flycheck.)

I tried quick-lint-js with Eglot/flymake, and everything is working as it should. I haven't tried the `LSP` version, but I assume it works fine too.

Flymake/flycheck is slow and problematic in my experience.

**Comment by strager** [comment]
I reproduced the problem. It seems like Flycheck isn't running quick-lint-js at all when I press C-x C-;.

**Comment by strager** [comment]
The bug doesn't happen with ESLint or pylint. I think it's an issue with quick-lint-js's integration.

**Comment by strager** [comment]
I think I found the solution. Edit your ~/.emacs:
```diff
-(setq-local flycheck-check-syntax-automatically '(mode-enabled idle-change))
+(setq-local flycheck-check-syntax-automatically '(mode-enabled idle-change new-line))
```
Can you try this and see if it works for you @cello-phane?

**Comment by cello-phane** [comment]
That fixes it, Thanks. I see that in the docs for flycheck it says new-line:
Check the buffer immediately after a new line was inserted.

So it seems the problem was the new-line getting inserted.

**Comment by strager** [comment]
Docs fixed in commit 3030a20f0004f4559f03c2892d30ffd90427f26e.
